File: Master1/Master1/ViewController.m
//
//  ViewController.m
//  Master1
//
//  Created by Davide Morelli on 07/11/14.
//  Copyright (c) 2014 unipi. All rights reserved.
//

#import "ViewController.h"
#import "Board.h"
#import "Dice.h"

@interface ViewController ()

@end

@implementation ViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    Board *board = [[Board alloc] init];
    board.listOfDice = [NSSet setWithArray:
                        [NSArray arrayWithObjects:
                         [[Dice alloc] init],
                         [[Dice alloc] init], nil]];
    
    NSLog(@"sum = %i", [board sumAll]);
    

    int mul = 10;
    
    void (^testBlock)(int) = ^(int a){
        NSLog(@"Integer is: %i", a * mul);
    };
    
    testBlock(1);
    mul = 20;

    void (^testBlock2)(int) = ^(int a){
        NSLog(@"Integer is: %i", a * mul);
    };

    testBlock2(1);

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: motion/motion/ViewController.swift
//
//  ViewController.swift
//  motion
//
//  Created by Davide Morelli on 30/01/15.
//  Copyright (c) 2015 BioBeats. All rights reserved.
//

import UIKit
import CoreMotion

class ViewController: UIViewController {

    var motionManager : CMMotionManager?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        motionManager = CMMotionManager()
        if let mm = motionManager
        {
            if mm.deviceMotionAvailable
            {
                NSLog("device available")
                if mm.gyroAvailable
                {
                    mm.gyroUpdateInterval = 1.0/10.0
                    mm.startGyroUpdatesToQueue(NSOperationQueue(), withHandler: { (gyroData, error) -> Void in
                        NSLog("x=%f y=%f z=%f",
                            gyroData.rotationRate.x,
                            gyroData.rotationRate.y,
                            gyroData.rotationRate.z)
                        
                    })
                }

            } else
            {
                NSLog("device NOT available")
            }
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

